libs/common/src/mongo-db-client/index.ts

```diff
@@ -0,0 +1,71 @@
+import { Collection, Db as DbConnection, MongoClient, WithoutId } from 'mongodb';
+import { MongoDbCollectionName } from '../types/mongo-db-models/mongo-db-collection-name';
+import { DbModel } from '../types/mongo-db-models/db-model';
+
+type ConnectionOptions = {
+  dbConnectionString: string;
+  dbName: string;
+};
+
+type MongoDbClientConnection =
+  | {
+      isInitialised: false;
+    }
+  | {
+      isInitialised: true;
+      client: MongoClient;
+      connection: DbConnection;
+    };
+
+export class MongoDbClient {
+  private dbName: string;
+
+  private dbConnectionString: string;
+
+  private mongoDbClientConnection: MongoDbClientConnection = {
+    isInitialised: false,
+  };
+
+  constructor({ dbName, dbConnectionString }: ConnectionOptions) {
+    this.dbName = dbName;
+    this.dbConnectionString = dbConnectionString;
+  }
+
+  private async initClient(): Promise<MongoDbClientConnection & { isInitialised: true }> {
```

Comment:
should this be like getOrInitialiseClient()? Trying to think how for rest of file we can show if the client exists we get it otherwise we initialise?